### DISCUSSION TITLE
Redirect-rule is disabled, but only some redirected script keeps running

### DISCUSSION THREAD

**Discussion by ytomino** [uBlockOrigin/uBlock-issues#2653] [Q&A] [answered]
Hello.

I tried to disable the redirect-rules, added below into "My filters".

```
||googletagmanager.com/gtag/js$script,xhr,redirect=googletagmanager_gtm.js:5,badfilter
||pagead2.googlesyndication.com/pagead/js/adsbygoogle.js$script,xhr,redirect=googlesyndication_adsbygoogle.js:5,domain=~zipextractor.app,badfilter
```

or

```
@@||googletagmanager.com/gtag/js$script,xhr,redirect-rule,domain=(SOME DOMAIN)
@@||pagead2.googlesyndication.com/pagead/js/adsbygoogle.js$script,xhr,redirect-rule,domain=(SOME DOMAIN)
```

with ```* * 3p-script block``` in dynamic filtering rules like "medium mode" in Wiki.

These both seem to be valid and the records in Logger have changed as I intended.

However I looked in the Firefox script debugger and found that some scripts continued to run. Breakpoints, etc. also work.

Apparently it depends on the combination of site and script, for example at https://crepu.net/ both scripts stopped, but at https://dragonball-neta.com/ only googletagmanager stopped.
Curiously, Logger records seem that the redirect rules are correctly disabled in both cases.

**Before adding the filters**:

![Screenshot_20230512_222951](https://github.com/uBlockOrigin/uBlock-issues/assets/442860/447a6791-a7ef-4f61-9e2b-4dd080b27b46)

The redirected scripts of googletagmanager and googlesyndication are running.

**After adding the filters**:

![Screenshot_20230512_223112](https://github.com/uBlockOrigin/uBlock-issues/assets/442860/c310ab0d-477b-49a3-a26b-f305c384a886)

googletagmanager is stopped, but redirected-version googlesyndication is still running.
However Logger seem that both scripts are not redirected.

I tried clearing the cache and ctrl+F5 but the result was the same.

Perhaps this only occurs in my environment.
Can anyone else reproduce this behavior?

Thanks.

Firefox 112.0.2
uBlock Origin 1.49.2
MX 21.3 Wildflower / Debian GNU/Linux 11 (bullseye)

**Comment by gorhill** [comment, marked as answer]
Firefox Enhanced Tracking protection has its own redirections, they are called _shim_. I think this is what you are seeing.

**Comment by ytomino** [comment]
Oh! Turning off Enhanced Tracking Protection also stopped googlesyndication.

![Screenshot_20230513_000406](https://github.com/uBlockOrigin/uBlock-issues/assets/442860/0e84923f-82a9-47f6-9bc5-e0f4d223c649)

I did not know that Firefox does this by default.
In the initial state, two different redirected scripts were being loaded at the same time.
I should have looked more carefully before asking the question.

Thank you for your quick answer!
